include/envoy/http/header_map.h

```diff
@@ -111,6 +111,12 @@ class HeaderString {
   void setInteger(uint64_t value);
 
   /**
+   * Set the value of the string to a static string reference.
+   * @param static_value MUST point to static data.
```

Comment:
How static is static here? From your review replies and comments below, I'm starting to think static isn't static as in the C/C++ allocation sense but it's more "this doesn't change for the lifetime of the referencing header map". Is this the correct reading?